components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -200,14 +417,309 @@ class BraveRewardsBrowserTest : public InProcessBrowserTest {
+    bool use_panel = true;
+    ClaimGrant(use_panel);
+
+    // Navigate to a verified site in a new tab
+    GURL url = embedded_test_server()->GetURL(publisher, "/index.html");
+    ui_test_utils::NavigateToURLWithDisposition(
+        browser(), url, WindowOpenDisposition::NEW_FOREGROUND_TAB,
+        ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
+
+    // Open the Rewards popup
+    content::WebContents* popup_contents = OpenRewardsPopup();
+    ASSERT_TRUE(popup_contents);
+
+    // Construct an observer to wait for the site banner to load
+    content::WindowedNotificationObserver site_banner_observer(
+        content::NOTIFICATION_LOAD_COMPLETED_MAIN_FRAME,
+        content::NotificationService::AllSources());
+
+    // Click button to initiate sending a tip
+    ASSERT_TRUE(ExecJs(
+        popup_contents,
+        "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+        "delay(0).then(() => "
+        "  document.querySelector(\"[type='tip']\").click());",
+        content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+        content::ISOLATED_WORLD_ID_CONTENT_END));
+
+    // Wait for the site banner to load
+    site_banner_observer.Wait();
+
+    // Retrieve the notification source
+    const auto& site_banner_source =
+        static_cast<const content::Source<content::WebContents>&>(
+            site_banner_observer.source());
+
+    content::WebContents* site_banner_contents = site_banner_source.ptr();
+    ASSERT_TRUE(site_banner_contents);
+
+    // Select the tip amount (1 BAT)
+    // TODO(emerick): Better to do this by data-test-id
+    ASSERT_TRUE(ExecJs(
+        site_banner_contents,
+        "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+        "delay(0).then(() => "
+        "  document.getElementsByTagName('button')[1].click());",
+        content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+        content::ISOLATED_WORLD_ID_CONTENT_END));
+
+    // Make this tip monthly, if requested
+    if (monthly) {
+      ASSERT_TRUE(ExecJs(
+          contents(),
+          "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+          "delay(0).then(() => "
+          "  document.querySelector(\"[data-test-id='monthlyCheckbox']\")"
+          "    .click());",
+          content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+          content::ISOLATED_WORLD_ID_CONTENT_END));
+    }
+
+    // Send the tip
+    // TODO(emerick): Better to do this by data-test-id
+    ASSERT_TRUE(ExecJs(
+        site_banner_contents,
+        "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+        "delay(0).then(() => "
+        "  document.getElementsByTagName('button')[4].click());",
+        content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+        content::ISOLATED_WORLD_ID_CONTENT_END));
+
+    // Wait for thank you banner to load
+    ASSERT_TRUE(WaitForLoadStop(site_banner_contents));
+
+    // Make sure that thank you banner shows correct publisher data
+    // (domain and amount)
+    {
```

Comment:
Could we do another check here?

If monthly, then check if `Monthly` string is there. This way we know that correct overlay is displayed, one time vs recurring.

![image](https://user-images.githubusercontent.com/9574457/54993466-2389e480-4fc2-11e9-964f-ac312e3e4c26.png)